rails-snow-queen/app/assets/javascripts/map.js

```diff
@@ -0,0 +1,110 @@
+class Map {
+	constructor(config) {
+    this.map = this.initMap();
+    this.drawingManager = this.initDrawingManager();
+    this.removeControl = this.initRemoveControl();
+    this.geocoder = this.initGeocoder();
+    this.marker = null;
+    this.geocodedAddress = null;
+    this.polygons = [];
+    this.onGeocodingResponse = config.onGeocodingResponse;
+    this.onPolygonsCreated = config.onPolygonsCreated;
+
+    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+    this.handleGeocodingResponse = this.handleGeocodingResponse.bind(this);
+    this.handlePolygonCreated = this.handlePolygonCreated.bind(this);
+    this.handleRemovePolygon = this.handleRemovePolygon.bind(this);
+    this.addListeners()
+  }
+  
+  initMap() {
+    return new google.maps.Map(document.getElementById('map'), {
+      zoom: 14,
```

Comment:
There are a lot of magic values here. It's not a priority right now, but eventually, it would be nice if they were separated out into possibly a `config.js` file or just in this file at the top. In a separate file, you could just import them without the need to clutter up this `map` class and you could add comments for each value to justify your change for them. In my experience, it's very common to not remember why you chose a certain location or zoom number months from now when you come back to this. 

```javascript
// config.ts
const constants = {
  // we are using this zoom level because blah blah blah…
  MAP_ZOOM_LEVEL: 14,
}

export { constants };
```

```javascript
\\ map.js
import { constants };
```

Reply:
Done